libs/spark/src/Unstable_Typography/Unstable_Typography.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import {
+  default as MuiTypography,
+  TypographyProps as MuiTypographyProps,
+} from '@material-ui/core/Typography';
+import makeStyles from '../makeStyles';
+import { OverrideProps } from '../utils';
+import { Unstable_TypographyVariant } from '../theme/unstable_typography';
+
+export interface Unstable_TypographyTypeMap<
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  P = {},
+  D extends React.ElementType = 'span'
+> {
+  props: P &
+    Omit<MuiTypographyProps, 'classes' | 'variant' | 'color'> & {
+      variant?: Unstable_TypographyVariant;
+      color?: 'initial' | 'inherit' | 'default' | 'inverse';
+    };
+  defaultComponent: D;
+  classKey: Unstable_TypographyClassKey;
+}
+
+export type Unstable_TypographyProps<
+  D extends React.ElementType = Unstable_TypographyTypeMap['defaultComponent'],
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  P = {}
+> = OverrideProps<Unstable_TypographyTypeMap<P, D>, D>;
+
+export type Unstable_TypographyClassKey = 'root';
+
+const useStyles = makeStyles<Unstable_TypographyClassKey>(
+  (theme) => ({
+    root: (props: Unstable_TypographyProps) => ({
+      ...(props.variant === 'display' && {
+        ...theme.unstable_typography.display,
+        color: theme.unstable_palette.text.heading,
+        ...(props.color === 'inverse' && {
+          color: theme.unstable_palette.text.inverseHeading,
+        }),
+      }),
+      ...(props.variant === 'T32' && {
+        ...theme.unstable_typography['T32'],
```

Comment:
Maybe this is naive but could you do this:
```suggestion
      ...{theme.unstable_typography[props.variant ],
```

My guess is that does not allow for good type-checking and if a value was passed in that did not match it would blow up.

Reply:
You could actually, but yes the typing would complain. Also we need it anyways for variant-specific colors.